### PR TITLE
Implement Start and Stop of Services

### PR DESCRIPTION
**Pull request by schrieveslaach**
In order to test the behaviour of services when resources are not available, PREvant now supports the status change of services. Therefore, PREvant offers a new resource which allows to start and stop containers. Additionally, the UI provides buttons, utilizing this resource.

Additionally, this commit cleans up the backend structure and it also uses the store to reflect the change to the application.

### CODE REVIEW

File: frontend/src/store/index.js [hunk 1]
```diff
@@ -123,6 +123,11 @@ export default new Vuex.Store( {
          } );
       },
 
+      updateServiceStatus( state, { appName, serviceName, serviceStatus } ) {
+         const service = state.apps[appName].find(service => service.name == serviceName);
+         Vue.set( service.state, 'status', serviceStatus );
```
Comment (x1B) [inline frontend/src/store/index.js]: Should this attribute not be present from the start? What if the service has already be stopped by someone else when loading the UI? But if it _is_ present from the start, `Vue.set` should not be necessary.

File: api/src/infrastructure/docker.rs [hunk 2]
```diff
@@ -641,6 +645,39 @@ impl Infrastructure for DockerInfrastructure {
...
+                let mut runtime = Runtime::new()?;
+                let details = runtime.block_on(c.inspect())?;
+                match status {
+                    ServiceStatus::Running => {
+                        if !details.state.running {
+                            runtime.block_on(c.start())?;
+                        }
+                    }
+                    ServiceStatus::Paused => {
+                        if details.state.running {
```
Comment (x1B) [inline api/src/infrastructure/docker.rs]: Does this mean that updates to the container could be lost, if the Web UI is toggled twice in quick succession, before Docker has implemented the change?

File: api/res/openapi.yml [hunk 3]
```diff
@@ -125,17 +125,40 @@ paths:
...
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/State'
+      responses:
+        '202':
+          description: The state change has been accepted
+        '404':
+          description: Cannot find app
```
[the remark this answers is not in the document]
Comment (schrieveslaach) [reply api/res/openapi.yml]: That's correct. I'll fix the description.

File: frontend/src/store/index.js (hunk 1 above)
Comment (schrieveslaach) [reply frontend/src/store/index.js]: That's true. I'll fix this.

File: api/src/infrastructure/docker.rs (hunk 2 above)
Comment (schrieveslaach) [reply api/src/infrastructure/docker.rs]: The code now checks if Docker's REST API return 304 and if so the frontend will received also 202. Other error will be propagated through an 500.

**Pull request merged**